server/core/src/main/java/org/infinispan/server/core/transport/NonRecursiveEventLoopGroup.java

```diff
@@ -34,6 +35,9 @@ public NonRecursiveEventLoopGroup(MultithreadEventLoopGroup eventLoopGroup) {
          throw new IllegalArgumentException("Provided multi threaded event loop group must have at least 2 executors, only has " + executors);
       }
       this.eventLoopGroup = eventLoopGroup;
+
+      // Set each of the event loop as the local executor
+      eventLoopGroup.forEach(ee -> ee.submit(() -> LocalExecutorThreadLocal.setLocalExecutor(ee)));
```

Comment:
It would be great if the even loops are distributed evenly instead of passing the group one

Reply:
@franz1981 not sure I understand what you mean here. Are you talking about the `EventExecutorChooserFactory.EventExecutorChooser` ? This method is exactly what I need afaik to set a thread local for each `EventLoop` thread.

Reply:
Sorry I misread what the code was doing, but still, it looks redundant: search for 
`ThreadExecutorMap.currentExecutor` - in theory Netty already allow to have a per-event-loop executor hook to retrieve the current event loop bound to a specific Thread - which is usually set on single event loop creation by Netty itself

Reply:
Yes, that is exactly what I was looking for originally! I will keep that mind if I resurrect this PR later, thanks again @franz1981 !